lib/Stats.js

```diff
@@ -9,29 +9,54 @@ const SizeFormatHelpers = require("./SizeFormatHelpers");
 const formatLocation = require("./formatLocation");
 const identifierUtils = require("./util/identifier");
 
-const optionsOrFallback = (...args) => {
-	let optionValues = [];
-	optionValues.push(...args);
+// XXX: types
+const Compilation = require("./Compilation");
```

Comment:
The following definitions don't work as expected:

```js
/** @typedef {import("./Compilation")} Compilation */
/** @typedef {typeof import("./Compilation")} Compilation */

Reply:
This is the main motivation of this PR: discover how to properly import our types

Reply:
`/** @typedef {typeof import("./Compilation")} Compilation */` is the right syntax I think. Apparently this won't work when the module itself has type errors

Reply:
I want this to be figured out, before merging this.

Adding additional requires could create dependency cycles which can lead to errors...